src/mscorlib/src/System/Runtime/MemoryFailPoint.cs

```diff
@@ -324,6 +324,9 @@ private static void CheckForAvailableMemory(out ulong availPageFile, out ulong t
         [System.Security.SecurityCritical]  // auto-generated
         private static unsafe bool CheckForFreeAddressSpace(ulong size, bool shouldThrow)
         {
+#if FEATURE_PAL // Remove this when GlobalMemoryStatusEx is able to provide legitimate estimates
+            return true;
```

Comment:
You are right about MemFreeAfterAddress is useless here, the only usage of it is in the above function CheckForFreeAddressSpace. That is the reason i am exiting early here as we will be stuck in an infinite loop in MemFreeAfterAddress as you have pointed out .

GlobalMemoryStatusEx returns the value for TopOfMemory, which is another artificial constraint that we currently say is 128TB, that is what i meant in my comment

@janvorli  do you think a better change would be to make MemFreeAfterAddress for FEATURE_PAL to always return TopOfMemory?